src/catalog/src/local/memory.rs

```diff
@@ -355,6 +359,34 @@ impl MemoryCatalogManager {
         let _ = manager.register_table_sync(request).unwrap();
         manager
     }
+
+    #[cfg(any(test, feature = "testing"))]
+    pub async fn try_new_with_table(table: TableRef) -> Result<Arc<Self>> {
```

Comment:
There's already a method named `new_with_table` for testing, let's reuse that. Change it if you wish. It's for testing purpose, simply "unwrap" all the way down.